core/src/main/scala/scalaz/Maybe.scala

```diff
@@ -390,6 +390,24 @@ sealed abstract class MaybeInstances extends MaybeInstances0 {
 
       override def filter[A](fa: Maybe[A])(f: A => Boolean): Maybe[A] =
         fa.filter(f)
+
+      override def alt[A](a1: => Maybe[A], a2: => Maybe[A]): Maybe[A] = altly2(a1, a2) {
```

Comment:
since `altly2` is for perf optimisation, you need to implement `alt` directly. It should be pretty easy, just evaluate the first one and if it is non-`Empty` return it, otherwise return the second one.

Reply:
You're right forgot to revert to the first impl.